### PR TITLE
Add isDirty to validator context

### PR DESCRIPTION
**Pull request by ArrayKnight**

### CODE REVIEW

**Comment by selbekk** [comment]
Hi! This looks great! There’s a few details I’d like to discuss, but I don’t have access to my computer until tuesday. 

For now, a thing I see is missing is documenting this feature. Do you want to look into this as well?

**Comment by ArrayKnight** [comment]
I've updated the documentation as well as made some minor cleanup edits.

File: README.md [hunk 1]
```diff
@@ -456,7 +473,10 @@ The `onSubmit` function is called with an object with the following props:
 {
     errors: object, // Object with all error messages, keyed per field
     fields: object, // Object with all field inputs, keyed per field
+    isDirty: object, // Object with all fields isDirty state, keyed per field
```
[the remark this answers is not in the document]
Comment (ArrayKnight) [reply README.md]: How about `areDirty`?

[the remark this answers is not in the document]
Comment (selbekk) [reply README.md]: That's very clever, but I think we need to be a bit careful about confusing the user as well. I'd rather be overly explicit than clever in the API-layer.

File: src/Form.js [hunk 2]
```diff
@@ -3,27 +3,25 @@ import { func, shape } from 'prop-types';
...
-        .filter(([fieldName]) => !fieldsToRemove.includes(fieldName))
-        .reduce(
-            (all, [fieldName, fieldValue]) => ({
-                ...all,
-                [fieldName]: fieldValue,
-            }),
-            {},
-        );
+const propTypes = {
+    onChange: func,
```
[the remark this answers is not in the document]
Comment (ArrayKnight) [reply src/Form.js]: This is less a feature, more just maintaining the ability to listen to the form's onChange event. It's optional and has no real tie in with the form logic.

[the remark this answers is not in the document]
Comment (selbekk) [reply src/Form.js]: fair enough I guess.

**Comment by ArrayKnight** [comment]
@selbekk Would you be able to guide me to the desired git commands?

I think you're asking to have the branch renamed to `feat/isDirty`.

But this is not something I've had to do and I'd prefer to get it done correctly and the way you'd like.

**Comment by ArrayKnight** [comment]
@selbekk I think I've accomplished what you're asking for? Can you please verify?

**Comment by selbekk** [comment]
@ArrayKnight let me see if there's something I can do at my end so you don't have to. You've done a great deal of work already! 😸

**Comment by selbekk** [comment]
Please let me know if you're interested in co-maintaining this with me. Reach out on Twitter :) twitter.com/selbekk

**Pull request merged**